src/rdkafka_transport.c

```diff
@@ -498,6 +500,14 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 	if (!SSL_set_fd(rktrans->rktrans_ssl, rktrans->rktrans_s))
 		goto fail;
 
+	mtx_lock(&rkb->rkb_lock);
+	rd_snprintf(name, sizeof(name), "%s", rkb->rkb_nodename);
+	mtx_unlock(&rkb->rkb_lock);
+	if ((t = strrchr(name, ':')))
+		*t = '\0';
+	if (!SSL_set_tlsext_host_name(rktrans->rktrans_ssl, name))
```

Comment:
👍  on #if check

Maybe it is safest not to enable this by default but provide a configuration option, this way we ensure not breaking functionality for existing users, and making the option explicit also allows us to document the hostname recommendation/requirement with the configuration option.
What do you think?

Otherwise, regarding host/ip, we'll need to dig up some RFCs to know exactly what constitutes a non-hostname (is digit . :  enough?)